addons/l10n_cl/models/res_partner.py

```diff
@@ -44,6 +44,8 @@ def _format_vat_cl(self, values):
 
     def _format_dotted_vat_cl(self, vat):
         vat_l = vat.split('-')
+        if len(vat_l) < 2:
+            raise ValidationError(_('The VAT number %s for partner %s does not seem to be valid.', vat, self.name))
```

Comment:
@DjamelTouati I think this method is mainly used in the QWeb of the report and raising there might not be ideal.   The write method has not been there all the time and maybe indeed, there are too many cases where you could go around it.  

Just a suggestion, but can not we change the method so it works in all cases?  @Danisan

Reply:
@jco-odoo yes indeed, this ValidationError is not a good idea and will not be thrown, because it is not catch since we call the function `_format_dotted_vat_cl` in a template.

However, this is still a problem, let's imagine you create a `" res.partner "` with a valid Chilean VAT number for example "CL22060449-7" and country "chilie" but you do not set the field "'l10n_latam_identification_type_id'".

- Here in the create function of `l10n_cl`:
https://github.com/odoo/odoo/blob/14.0/addons/l10n_cl/models/res_partner.py#L53

  the function `" _format_vat_cl "` will be called in order to format the vat, but it needs the field `"l10n_latam_identification_type_id "` so the vat will not be formatted: https://github.com/odoo/odoo/blob/14.0/addons/l10n_cl/models/res_partner.py#L38-L43

- Then here, the `”_fix_vat_number”` function will be called:
https://github.com/odoo/odoo/blob/14.0/addons/base_vat/models/res_partner.py#L553

  As the `_format_vat_cl` function is private so the format_func variable will have the value of `"stdnum_vat_fix_func"` so the `"compact"` function which will remove the dash from “CL22060449-7” the result will be “CL220604497” which is an invalid Chilean vat number: https://www.fonoa.com/countries/chile#:~:text=Chilean%20VAT%20number%20contains%207,567%2D8

- The `" format_dotted_vat_cl "` function needs a valid Chilean vat number, so with a dash in the middle in order to split the result in two:
https://github.com/odoo/odoo/blob/14.0/addons/l10n_cl/models/res_partner.py#L46-L47